source/MaterialXRuntime/Commands/PrimCommands.cpp

```diff
@@ -35,6 +35,27 @@ void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& path, R
 
 void createPrim(RtStagePtr stage, const RtToken& typeName, const RtPath& parentPath, const RtToken& name, RtCommandResult& result)
 {
+    static RtTokenSet LOOK_MANAGEMENT_NODES { RtToken("collection"),
+                                              RtToken("look"),
+                                              RtToken("lookgroup"),
+                                              RtToken("materialassign")
+                                            };
+    static RtTokenSet MATERIAL_NODES { RtToken("surfacematerial"),
+                                       RtToken("volumematerial")
+                                     };
+    RtPrim nodeDefPrim =
+        RtApi::get().getMasterPrim(RtToken(typeName));
+
+    // If the node does not have a NodeDef it's either a look management node or a material node
+    if (!nodeDefPrim) {
+        if (LOOK_MANAGEMENT_NODES.count(typeName) == 0 && MATERIAL_NODES.count(typeName) == 0) {
```

Comment:
I think it would be more flexible and efficient to let each schema decide if the corresponding prim is creatable inside a specific parent or not. Each registered schema has a creator function, which gets the parent where it is about to be created. So it's easy to check if the parent is a valid parent of not there.

For example see RtLook::createPrim(), RtNode::createPrim(), RtNodeGraph::createPrim(), etc. each of these can decide if the given parent is a valid parent or not for the corresponding prim type (throw if the parent is invalid).

I guess most of them only allow to be created at root level (all except nodes) so you can check if the parent is root or not. And for nodes you can check that the parent is a nodegraph.